x-pack/plugins/security_solution/public/management/pages/trusted_apps/store/selectors.ts

```diff
@@ -24,12 +25,15 @@ const pageInfosEqual = (pageInfo1: PaginationInfo, pageInfo2: PaginationInfo): b
 export const needsRefreshOfListData = (state: Immutable<TrustedAppsListPageState>): boolean => {
   const currentPageInfo = state.listView.currentPaginationInfo;
   const currentPage = state.listView.currentListResourceState;
+  const freshDataTimestamp = state.listView.freshDataTimestamp;
 
   return (
     state.active &&
-    isOutdatedResourceState(currentPage, (data) =>
-      pageInfosEqual(currentPageInfo, data.paginationInfo)
-    )
+    isOutdatedResourceState(currentPage, (data) => {
+      return (
+        pageInfosEqual(currentPageInfo, data.paginationInfo) && data.timestamp >= freshDataTimestamp
```

Comment:
For my understanding, the timestamp comparison is meant to decide when to refresh primarily in the case where we update data like in deletions or creations?  I like the concept as it gives us another bit of control for when we update data.